docs/src/symbolic_regression.md

```diff
@@ -0,0 +1,137 @@
+# Symbolic Regression
```

Comment:
I would expect `DirectProblem` to be documented here. OccamNet is far from the only symbolic regression algorithm: all of the others are too!

Reply:
It uses the problem further down. I will add some more documentation for this once https://github.com/MilesCranmer/SymbolicRegression.jl/pull/37 is done and SymbolicRegression is interfaced.

Reply:
But we do not need to solely rely on the DirectProblem. We can also use it to replace SINDy etc. It just depends on the target switching from Y = somethingsomething to DX = somethingelse or X[:, 2:end] = somethingelseentirely.

Reply:
Alright, I would expect a good section on things like `STLQS` and `ImplicitOptimizer` describing them as sparse regression algorithms as well. Having the 3 different classes of methods, OccamNet, SparseRegression.jl, and the basis library methods, all in one comprehensive API, is going to be awesome.

Reply:
And maybe we can get to #160 and #192